kong/utils.go

```diff
@@ -103,3 +111,27 @@ func HTTPClientWithHeaders(client *http.Client,
 	}
 	return res
 }
+
+func ParseSemanticVersion(v string) (semver.Version, error) {
+	// fix enterprise edition semver adding patch number
+	// fix enterprise edition version with dash
+	// fix bad version formats like 0.13.0preview1
+	re := regexp.MustCompile(`(\d+\.\d+)(?:[\.-](\d+))?(?:\-?(.+)$|$)`)
+	m := re.FindStringSubmatch(v)
+	if len(m) != versionParts {
+		return semver.Version{}, fmt.Errorf("Unknown Kong version : '%v'", v)
+	}
+	if m[2] == "" {
+		m[2] = "0"
+	}
+	if m[3] != "" {
+		m[3] = "-" + strings.Replace(m[3], "enterprise-edition", "enterprise", 1)
+		m[3] = strings.Replace(m[3], ".", "", -1)
+	}
+	v = fmt.Sprintf("%s.%s%s", m[1], m[2], m[3])
+	return semver.Make(v)
+}
+
+func versionFromInfo(info map[string]interface{}) string {
```

Comment:
Please export this function so that it can be used by end users to extract version.
```suggestion
func VersionFromInfo(info map[string]interface{}) string {
```